### PR TITLE
Added support for wagtail 4.0

### PR DESCRIPTION
**Pull request by katdom13**
Release notes: https://docs.wagtail.org/en/stable/releases/4.0.html

- Added test for wagtail 4
- Updated documentation
- Suppressed some warnings for wagtail 4

Tests run OK
```
Found 73 test(s).
Creating test database for alias 'default'...
Adding content type 'wagtail_transfer | idmapping'
Adding content type 'wagtail_transfer | importedfile'
System check identified some issues:

WARNINGS:
?: (wagtailadmin.W003) The WAGTAILADMIN_BASE_URL setting is not defined
        HINT: This should be the base URL used to access the Wagtail admin site. Without this, URLs in notification emails will not display correctly.

System check identified 1 issue (0 silenced).
.../Users/kdomingo/Desktop/torchbox/forks/wagtail-transfer/.venv/lib/python3.9/site-packages/djangorestframework-3.14.0-py3.9.egg/rest_framework/pagination.py:200: UnorderedObjectListWarning: Pagination may yield inconsistent results with an unordered object_list: <class 'tests.models.Category'> QuerySet.
  paginator = self.django_paginator_class(queryset, page_size)
......................................................................
----------------------------------------------------------------------
Ran 73 tests in 1.514s

OK
```

### CODE REVIEW

File: tests/tests/test_import.py [hunk 1]
```diff
@@ -123,15 +123,24 @@ def test_import_pages(self):
         self.assertEqual(updated_page.draft_title, "New home")
 
         # get_latest_revision (as used in the edit-page view) should also reflect the imported content
-        updated_page_revision = updated_page.get_latest_revision_as_page()
+        if WAGTAIL_VERSION >= (4, 0):
```
Comment (nickmoreton) [inline tests/tests/test_import.py:126]: Yep I like it 👍

File: tests/tests/test_import.py [hunk 2]
```diff
@@ -123,15 +123,24 @@ def test_import_pages(self):
...
         self.assertEqual(updated_page_revision.intro, "This is the updated homepage")
         self.assertEqual(updated_page_revision.title, "New home")
 
         created_page = SimplePage.objects.get(url_path='/home/imported-child-page/')
         self.assertEqual(created_page.intro, "This page is imported from the source site")
         # An initial page revision should also be created
         self.assertTrue(created_page.get_latest_revision())
-        created_page_revision = created_page.get_latest_revision_as_page()
+
+        if WAGTAIL_VERSION >= (4, 0):
```
Comment (nickmoreton) [inline tests/tests/test_import.py:139]: Ditto

**Review by nickmoreton — APPROVED** [review]
Thanks @katdom13 These changes look good

**Comment by jacobtoppm** [comment]
Hi @katdom13, did you test this manually at all or just use the unit tests?

**Comment by katdom13** [comment]
Hi @jacobtoppm ,
May I know how to test this manually? Thanks!

**Comment by jacobtoppm** [comment]
Hi @katdom13. You will need to set up two local instances of Wagtail, running at the same time, on different ports, with different databases - eg 8000 and 8001 - and with this package installed. I typically use bakerydemo as the demo site for easy setup, for example running one using https://github.com/wagtail/vagrant-wagtail-develop and one in a virtualenv. You will then need to set up the sites to import from/export to the other, following the instructions for the various settings. You should then check you can import both pages and snippets.

Sorry to be a pain on this one, but with complicated packages like Transfer, it's vital we do at least some manual testing - because the package requires two sites to talk to one another, the unit tests heavily rely on mocks and fixture data, which means we should really do more integration testing alongside them.

**Comment by katdom13** [comment]
Hi @jacobtoppm ,
This has now been manually tested and I can confirm that it works.
Steps taken:
- Setup two instances of Wagtail bakery demo, with one instance running on Docker, the other on Virtualenv.
- Edit some pages on the Docker instance
- Import the pages on the Virtualenv.
- Imports were successful.
Please let me know if there's anything else I can do for this PR.

Thank you!

**Pull request merged**